src/analysis/processing/qgsalgorithmvectorize.cpp

```diff
@@ -113,7 +113,7 @@ QVariantMap QgsVectorizeAlgorithmBase::processAlgorithm( const QVariantMap &para
       for ( int column = 0; column < iterCols; column++ )
       {
         const double value = rasterBlock->valueAndNoData( row, column, isNoData );
-        if ( !isNoData )
+        if ( !isNoData && !qgsDoubleNear( value, 0.0 ) )
```

Comment:
What's this for? Looks unrelated

Reply:
It is unrelated, I'll talk to you about that one later :wink: